Fix code quality issues

withshubh opened the pull request:
This pull request fixes some of the code quality issues raised by DeepSource on my fork of this repository.
I have already fixed some issues using DeepSource's Autofix.

Take a quick look at all the issues caught by DeepSource for this repository [here](https://deepsource.io/gh/withshubh/monkey/issues/?category=recommended)

### Summary of fixes

- Remove re-imported module
- Remove unused imports
- Use literal syntax instead of function calls to create data structure
- Use `is` to compare type of objects
- Remove unnecessary comprehension
- Remove length check in favour of truthiness of the object

You can also have a look at the [configuration file](https://github.com/withshubh/monkey/blob/deepfix/.deepsource.toml) I used for DeepSource Analysis.

acepace commented:
A lot of these commits look nice but they're *horribly mangled* by changing a lot of line breaks.

This is problematic for two reasons
* Screws up git history
* Not all the line breaks are...prettier looking.

Can you resubmit the PRs without the code formatting?

withshubh commented:
> A lot of these commits look nice but they're _horribly mangled_ by changing a lot of line breaks.
> 
> This is problematic for two reasons
> 
> * Screws up git history
> * Not all the line breaks are...prettier looking.
> 
> Can you resubmit the PRs without the code formatting?

Hi @acepace :wave: 
I'll resubmit the PR without the code formatting. :smile:

withshubh commented:
Hey @acepace :wave: 

I've updated the PR(fixes without the code formatting) but the CI check is failing since I removed the AutoPep8 transformer from the DeepSource configuration.

acepace commented:
Hm.. I'm not a github wiz. 
Can you try rebasing the PR so it never included that?

withshubh commented:
@acepace :star_struck: 
Please have a look :eyes:

shreyamalviya commented:
@withshubh, it's hard to see exactly what changes you're proposing since this PR contains changes that have already been merged into the 'develop' branch. Can you fetch the latest changes from 'develop', rebase your branch, and push again?
Feel free to ask if you've any questions on how to do this. :)

withshubh commented:
@shreyamalviya My forked develop is even with the upstream develop so now I'll push the fixes!

acepace commented on monkey/infection_monkey/utils/plugins/pluginTests/BadImport.py:
```diff
@@ -1,5 +1,3 @@
-from infection_monkey.utils.plugins.pluginTests.PluginTestClass import \
-    TestPlugin  # noqa: F401
 
```
Remove, it's there for a reason. Real test

withshubh replied:
Okay. I will remove this.

acepace commented on monkey/monkey_island/cc/resources/monkey.py:
Why is this better if we're straight converting to a list?

withshubh replied:
It is unnecessary to use a comprehension just to loop over the iterable and create a `list/set/dict` out of it. Python has a specialized set of tools for this task: the `list/set/dict `constructors, which are faster and more readable.
The same reason goes for the next two reviews.

shreyamalviya replied:
Build fails since this is under-indented.

acepace commented on monkey/monkey_island/cc/resources/monkey.py:
Why is this better if we're straight converting to a list?

shreyamalviya replied:
Build fails since this is under-indented.

acepace commented on monkey/monkey_island/cc/services/edge/displayed_edge.py:
```diff
@@ -63,7 +63,7 @@ def get_pseudo_label(edge: Dict):
     def services_to_displayed_services(services, for_report=False):
         if for_report:
-            return [x for x in services]
+            return list(services)
```
Why is this better if we're straight converting to a list?

mssalvatore replied:
I do think this more clearly shows the intent of the code. If I want to understand what's being returned, I'll understand what `list()` is doing faster than if I have to parse the list comprehension.

acepace reviewed (changes requested):
Most look good, some require explanation, one requires change.

withshubh commented:
> Most look good, some require explanation, one requires change.

@acepace The purpose of this PR was to show that how DeepSource can help in recognizing potential code quality issues. With DeepSource integrated into this repository you can automate analysis and fix issues using its auto-fix which generates the PR with fixes and for the issues like the one you mentioned above which are intentional or you don't decide to fix then you can choose to ignore them directly from the DeepSource dashboard.
Since DeepSource is free for Open Source I'd suggest you give it a try.

acepace commented:
> > Most look good, some require explanation, one requires change.
> 
> @acepace The purpose of this PR was to show that how DeepSource can help in recognizing potential code quality issues. With DeepSource integrated into this repository you can automate analysis and fix issues using its auto-fix which generates the PR with fixes and for the issues like the one you mentioned above which are intentional or you don't decide to fix then you can choose to ignore them directly from the DeepSource dashboard.
> Since DeepSource is free for Open Source I'd suggest you give it a try.

I agree. I'm not the maintainer anymore (used to be) but will check it out

withshubh commented:
Hey! @shreyamalviya I reverted the fix(remove unnecessary use of comprehension) that was breaking the tests.

withshubh commented:
Hi :wave:  @ShayNehmad Please have a look at this PR!

VakarisZ commented on .deepsource.toml:
```diff
@@ -0,0 +1,14 @@
+version = 1
+
+test_patterns = ["**/**_test.py"]
+
+[[analyzers]]
+name = "python"
+enabled = true
+dependency_file_paths = [
+  "monkey/monkey_island/requirements.txt",
+  "monkey/infection_monkey/requirements.txt"
+  ]
+
+  [analyzers.meta]
+  runtime_version = "3.x.x"
```
We don't need this file until we integrate deepsource. I think you should remove this file from this PR and instead make an issue called `Integration of deepsource` where you can put the contents of this file. Once we start deepsource integration, we'll pick it up.

VakarisZ reviewed (changes requested):
Nice fixes! Remove deepsource configuration file and we're good to go!

withshubh commented:
> Nice fixes! Remove deepsource configuration file and we're good to go!

Done! @VakarisZ :sparkles:

VakarisZ reviewed (approved):
Good job!
